File: app/src/main/java/uci/localproxy/util/fontAwesome/ButtonAwesome.java
package uci.localproxy.util.fontAwesome;

import android.content.Context;
import android.graphics.Typeface;
import android.util.AttributeSet;
import android.util.LruCache;

import androidx.appcompat.widget.AppCompatButton;


public class ButtonAwesome extends AppCompatButton {

	private final static String NAME = "FONTAWESOME";
	private static LruCache<String, Typeface> sTypefaceCache = new LruCache<String, Typeface>(12);

	
	public ButtonAwesome(Context context) {
		super(context);
		init();
			
	}
	public ButtonAwesome(Context context,AttributeSet attrs) {
		super(context,attrs);
		init();	
	}
	public ButtonAwesome(Context context, AttributeSet attrs, int defStyle){
		super(context,attrs,defStyle);
		init();	
	}
	
	public void init(){
		Typeface typeface = sTypefaceCache.get(NAME);

		if (typeface == null) {

			typeface = Typeface.createFromAsset(getContext().getAssets(), "fontawesome-webfont.ttf");
			sTypefaceCache.put(NAME, typeface);

		}

		setTypeface(typeface);
	}
}

File: app/src/main/java/grgr/localproxy/util/fontAwesome/ButtonAwesome.kt
package grgr.localproxy.util.fontAwesome

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import android.util.LruCache
import androidx.appcompat.widget.AppCompatButton

class ButtonAwesome : AppCompatButton {
    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) : super(
        context,
        attrs,
        defStyle
    ) {
        init()
    }

    fun init() {
        var typeface = sTypefaceCache[NAME]
        if (typeface == null) {
            typeface = Typeface.createFromAsset(context.assets, "fontawesome-webfont.ttf")
            sTypefaceCache.put(NAME, typeface)
        }
        setTypeface(typeface)
    }

    companion object {
        private const val NAME = "FONTAWESOME"
        private val sTypefaceCache = LruCache<String, Typeface?>(12)
    }
}